### PR TITLE
#16 Flush the object cache when switching databases.

### PR DESCRIPTION
**Pull request by OllieJones**
When activating or deactivating the plugin, it's necessary to flush the object cache as late as possible before switching databases.

This change uses PHP_INT_MAX for hook priorities to make sure the hooked functions are called very late.

### CODE REVIEW

File: deactivate.php [hunk 1]
```diff
@@ -47,12 +52,17 @@ function() {
 				$value = maybe_unserialize( $row->option_value );
 				if ( is_array( $value ) ) {
 					$value_flipped = array_flip( $value );
-					unset( $value_flipped[ str_replace( WP_PLUGIN_DIR . '/', '', SQLITE_MAIN_FILE ) ] );
+					$items         = array_reverse( explode( DIRECTORY_SEPARATOR, SQLITE_MAIN_FILE ) );
+					$item          = $items[1] . DIRECTORY_SEPARATOR . $items[0];
+					unset( $value_flipped[ $item ] );
```
Comment (aristath) [inline deactivate.php:57]: Why this change? 🤔

Comment (OllieJones) [reply deactivate.php:57]: This works correctly in environments where plugins are symlinked into `wp-content/plugins`. (And, of course, in environments where they're simply installed there.)

Comment (aristath) [reply deactivate.php:57]: Ah OK! Makes sense now. Thank you for the insights! 👍
